### PR TITLE
feat(v2): update create form modal design

### PR DESCRIPTION
**Pull request by karrui**
## Problem

This PR updates the design implementation of `CreateFormModal` to match new design.
The changes includes adding a confirmation checkbox and updating the buttons.
A new story for the storage mode ack screen has also been added for regression testing.

## Solution

**Features**:

- feat: email key using button click instead of link, update styling
- feat: add secret key ack before allowing storage form creation

**Improvements**:
- Extract components into their own file for cleanliness
- feat(CreateFormModal): reset modal state on close automatically
  - no more janky reset calls, the screen contents will now be unmounted when the modal is closed.

## Before & After Screenshots
- New story for storage mode ack has been added
- Existing create form modal story should have been updated to render the new design.

### CODE REVIEW

**Comment by karrui** [comment]
Converting to draft, implementing tag-creation input

File: frontend/src/features/workspace/components/CreateFormModal/CreateFormModalContent/SaveSecretKeyScreen.tsx [hunk 1]
```diff
@@ -102,17 +142,29 @@ export const SaveSecretKeyScreen = (): JSX.Element => {
...
+          <Box mb="2.5rem">
+            <Checkbox
+              aria-label="Storage mode form acknowledgement"
+              {...register('storageAck', {
+                required: true,
+              })}
+            >
+              If I lose my Secret Key, I cannot activate my form and all my
+              responses will be lost
+            </Checkbox>
```
Comment (timotheeg) [inline frontend/src/features/workspace/components/CreateFormModal/CreateFormModalContent/SaveSecretKeyScreen.tsx:166]: 👍  Jackson will be pleased 😄

File: frontend/src/features/workspace/components/CreateFormModal/CreateFormModalContent/CreateFormModalContent.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  useCreateFormWizard,
+} from '../CreateFormWizardContext'
+
+import { CreateFormDetailsScreen } from './CreateFormDetailsScreen'
+import { SaveSecretKeyScreen } from './SaveSecretKeyScreen'
+
+const SCREEN_ANIMATION_VARIANT = {
+  enter: (direction: number) => {
+    return {
+      x: direction > 0 ? 1000 : -1000,
```
Comment (timotheeg) [inline frontend/src/features/workspace/components/CreateFormModal/CreateFormModalContent/CreateFormModalContent.tsx:17]: Just checking: are there extra restrictions on what values `direction` can take?

**Review by timotheeg — APPROVED** [review]
lgtm 👍

File: frontend/src/features/workspace/components/CreateFormModal/CreateFormModalContent/CreateFormModalContent.tsx (hunk 2 above)
Comment (karrui) [reply frontend/src/features/workspace/components/CreateFormModal/CreateFormModalContent/CreateFormModalContent.tsx:17]: no restrictions tbh; framer will automatically extrapolate. I haven't tried 1:-1 to see if it affects anything.

Comment (timotheeg) [reply frontend/src/features/workspace/components/CreateFormModal/CreateFormModalContent/CreateFormModalContent.tsx:17]: Yeah, that's where I was going with my comment: if direction is a normal set (-1,1), then we can just multiply by the target position. Not very important though 😅

**Pull request merged**